src/MSBuild/PerformanceLogEventListener.cs

```diff
@@ -55,10 +55,16 @@ internal static PerformanceLogEventListener Create()
                 // Check to see if we should enable the event listener.
                 string logDirectory = Environment.GetEnvironmentVariable(PerfLogDirEnvVar);
 
-                if (!string.IsNullOrEmpty(logDirectory) && Directory.CreateDirectory(logDirectory).Exists)
+                if (!string.IsNullOrEmpty(logDirectory))
                 {
-                    eventListener = new PerformanceLogEventListener();
-                    eventListener.Initialize(logDirectory);
+                    if (Directory.CreateDirectory(logDirectory).Exists)
+                    {
+                        eventListener = new PerformanceLogEventListener();
+                        eventListener.Initialize(logDirectory);
+                    }
+
+                    // Delete env var so new starting processes will not duplicate Event listening
+                    Environment.SetEnvironmentVariable(PerfLogDirEnvVar, null);
```

Comment:
Do you have to reset this at the end of the build?

Reply:
On second, morning, thought I have reverted that change. I though that it was cross process event listener but it is actually in-process event listener which needs to run for every process involved.
TLDR: It was good as it was.